specs-go/chain/chainid.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package chain provides implementations of the ChainID calculation used in
+// identifying the result of subsequent layer applications.
+package chain
+
+import "github.com/docker/distribution/digest"
```

Comment:
We need to vendor this, right?

Or, we move that package into this repo. Given the strong dependence on that library and that these string digests are [part of the spec](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests-and-verification) seems we should add that package here as well. We already have one open-coded impl here: https://github.com/opencontainers/image-spec/blob/ebee8d2c7a1f3ea37a4ca0fed459db487456a16b/schema/manifest_backwards_compatibility_test.go#L113